diff -Naur darkplaces-rev20110628.orig/makefile darkplaces-rev20110628/makefile
--- darkplaces-rev20110628.orig/makefile	2010-08-27 18:48:41.000000000 +0200
+++ darkplaces-rev20110628/makefile	2012-05-10 13:51:54.000000000 +0200
@@ -75,6 +75,14 @@
 	# libjpeg dependency (set these to "" if you want to use dynamic loading instead)
 	CFLAGS_LIBJPEG=-DLINK_TO_LIBJPEG
 	LIB_JPEG=-ljpeg
+
+	# zlib dependency (set these to "" if you want to use dynamic loading instead)
+	CFLAGS_ZLIB=-DLINK_TO_ZLIB
+	LIB_ZLIB=-lz
+
+	# vorbis dependency (set these to "" if you want to use dynamic loading instead)
+	CFLAGS_LIBVORBIS=-DLINK_TO_LIBVORBIS
+	LIB_VORBIS=-lvorbisfile -lvorbis
 endif
 
 # Mac OS X configuration
diff -Naur darkplaces-rev20110628.orig/makefile.inc darkplaces-rev20110628/makefile.inc
--- darkplaces-rev20110628.orig/makefile.inc	2011-06-01 12:47:42.000000000 +0200
+++ darkplaces-rev20110628/makefile.inc	2012-05-10 13:50:13.000000000 +0200
@@ -187,7 +187,7 @@
 
 
 # Compilation
-CFLAGS_COMMON=$(CFLAGS_MAKEDEP) $(CFLAGS_PRELOAD) $(CFLAGS_FS) $(CFLAGS_WARNINGS) $(CFLAGS_LIBJPEG) $(CFLAGS_D3D) -D_FILE_OFFSET_BITS=64 -D__KERNEL_STRICT_NAMES
+CFLAGS_COMMON=$(CFLAGS_MAKEDEP) $(CFLAGS_PRELOAD) $(CFLAGS_FS) $(CFLAGS_WARNINGS) $(CFLAGS_LIBJPEG) $(CFLAGS_LIBVORBIS) $(CFLAGS_ZLIB) $(CFLAGS_D3D) -D_FILE_OFFSET_BITS=64 -D__KERNEL_STRICT_NAMES
 CFLAGS_DEBUG=-ggdb
 CFLAGS_PROFILE=-g -pg -ggdb -fprofile-arcs
 CFLAGS_RELEASE=
@@ -218,7 +218,7 @@
 
 OBJ_GLX= builddate.c sys_linux.o vid_glx.o thread_pthread.o keysym2ucs.o $(OBJ_SOUND) $(OBJ_CD) $(OBJ_COMMON)
 
-LDFLAGS_UNIXCOMMON=-lm $(LIB_ODE) $(LIB_JPEG) $(LIB_CRYPTO) $(LIB_CRYPTO_RIJNDAEL)
+LDFLAGS_UNIXCOMMON=-lm $(LIB_ODE) $(LIB_JPEG) $(LIB_VORBIS) $(LIB_ZLIB) $(LIB_CRYPTO) $(LIB_CRYPTO_RIJNDAEL)
 LDFLAGS_UNIXCL=-L$(UNIX_X11LIBPATH) -lX11 -lXpm -lXext -lXxf86dga -lXxf86vm -pthread $(LIB_SOUND)
 LDFLAGS_UNIXCL_PRELOAD=-lz -ljpeg -lpng -logg -ltheora -lvorbis -lvorbisenc -lvorbisfile -lcurl -lmodplug
 LDFLAGS_UNIXSV_PRELOAD=-lz -ljpeg -lpng -lcurl
